docs/dom-testing-library/api-async.md

```diff
@@ -35,6 +35,8 @@ await waitFor(() => expect(mockAPI).toHaveBeenCalledTimes(1))
 // ...
 ```
 
+⚠️  Using this utility with `expect.assertions` will result in flaky tests. Beware!
```

Comment:
If we were to add an explanation here, I would suggest explaining the exact problem, rather than saying it makes your test "flaky".

```suggestion
`waitFor` may run the callback a variable number of times
```

Reply:
Excellent suggestion